plan/logical_plan_builder.go

```diff
@@ -1800,10 +1812,20 @@ func (b *planBuilder) buildSemiJoin(outerPlan, innerPlan LogicalPlan, onConditio
 	if b.TableHints() != nil {
```

Comment:
can this branch be extracted as a function.

Reply:
@winoros this branch can be extracted as a function?

Reply:
i think it's not too much need to do this.